### PR TITLE
feat: Provide the capability to debug quarkus and qute ls (#769)

### LINKED ISSUE #769: Provide the capability to debug quarkus and qute ls
To improve LSP client for IJ we need to debug the MicroProfile LS or Qute LS. For instance we could add à breakpoint when the language server is initialized and see what happens on LSP client for IJ (to emulate à long start to initialize the language server).

Or we could set à breakpoint when diagnostic is published and update the IJ text editor to publish bad diagnostic which is unsynchronized with the document to publish diagnostic with bad offset.

In other words we could emulate corner usecase by setting breakpoint to stop the process from some opération and emuluate some slow operation.

I will work on this issue.

### PR DESCRIPTION
**Pull request by angelozerr**
Provide the capability to debug quarkus and qute ls

Fixes #769

### CODE REVIEW

File: src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/server/JavaProcessStreamConnectionProvider.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        if (!existsInPath) {
+            File f = new File(System.getProperty("java.home"),
+                    "bin/java" + (isWindows() ? ".exe" : ""));
+            javaPath = f.getAbsolutePath();
+        }
+        return javaPath;
+    }
+
+    private static boolean isWindows() {
+        return (OS.indexOf("win") >= 0);
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/server/JavaProcessStreamConnectionProvider.java:46]: Have you some suggestion please.

[the remark this answers is not in the document]
Comment (gastaldi) [reply src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/server/JavaProcessStreamConnectionProvider.java:46]: You can use `smallrye-common` or adapt to the same code as in https://github.com/smallrye/smallrye-common/blob/main/os/src/main/java/io/smallrye/common/os/OS.java

[the remark this answers is not in the document]
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/server/JavaProcessStreamConnectionProvider.java]: Thanks so much for your suggestion! I copied the OS enum and I use it now.

**Comment by angelozerr** [comment]
Ok I'm fixing that. Is it working for you?

**Comment by adietish** [comment]
works for me in IJ. To connect a debugger one needs to do Run > Attach to process and select the process listed with port 1044

**Review by adietish — APPROVED** [review]

**Pull request merged**